[addons] add settings support for property condition (basic conditions only)

AlwinEsch opened the pull request:
## Description
Before was it now possible to get a for Kodi defined condition on addon
settings. With this are the basic conditions also available for addon,
to define OS related parts.

The more complex conditions are not added, not sure if this goes then
to far to allow addons also this.

I'm not sure if this can in Leia or to wait for M*.

<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
Tested with this commit here:
https://github.com/xbmc/visualization.spectrum/pull/25/commits/cfef6c42ec5e9cf957862ad4b862ce0f43ca44b8
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

ksooo commented on xbmc/addons/settings/AddonSettings.cpp:
```diff
@@ -391,6 +392,13 @@ void CAddonSettings::InitializeControls()
+  // add basic conditions
+  const std::set<std::string> &simpleConditions = CSettingConditions::GetSimpleConditions();
+  for (std::set<std::string>::const_iterator itCondition = simpleConditions.begin(); itCondition != simpleConditions.end(); ++itCondition)
+    GetSettingsManager()->AddCondition(*itCondition);
```
```c++
for (const auto& condition : simpleConditions)
    GetSettingsManager()->AddCondition(condition);
```

AlwinEsch replied:
Yeah, becomes changed :-D, was a c&p from Kodi's system settings place ;-)

ksooo commented on xbmc/addons/settings/AddonSettings.cpp:
```diff
@@ -391,6 +392,13 @@ void CAddonSettings::InitializeControls()
+  CSettingConditions::Initialize();
+
+  // add basic conditions
+  const std::set<std::string> &simpleConditions = CSettingConditions::GetSimpleConditions();
```
```c++
const std::set<std::string>& simpleConditions = ...
```
align `&`to the type, please.

ksooo commented:
Could you please explain what this change is good for? What can an addon do after this change what was not possible before?

AlwinEsch commented:
See the link above, on a addon seetings.xml is a value like them, not possible:
```*.xml
        <setting id="pointsize" type="integer" label="30015">
          <default>3</default>
          <constraints>
            <minimum>1</minimum>
            <step>1</step>
            <maximum>50</maximum>
          </constraints>
          <control type="slider" format="integer">
            <formatlabel>30016</formatlabel>
          </control>
          <dependencies>
            <dependency type="enable" setting="mode" operator="is">2</dependency>
            <dependency type="visible">
              <or>
                <condition on="property" name="IsDefined">HAS_GL</condition>
                <condition on="property" name="IsDefined">HAS_GLES</condition>
              </or>
            </dependency>
          </dependencies>
        </setting>
```

For basic are the following by my system:
- 'has_ae_quality_levels'
- 'has_airplay'
- 'has_filesystem_smb'
- 'has_gl'
- 'has_upnp'
- 'has_web_server'
- 'has_zeroconf'
- 'have_lcms2'
- 'have_libva'
- 'have_libvdpau'
- 'have_x11'
- 'true'
- 'webserver_has_ssl'

This `property` values was not usable before on addon settings

ksooo commented:
Ah, okay. Thanks for explaining.

AlwinEsch committed:
[addons] add settings support for property condition (basic conditions only)

Before was it now possible to get a for Kodi defined condition on addon
settings. With this are the basic conditions also available for addon,
to define OS related parts.

The more complex conditions are not added, not sure if this goes then
to far to allow addons also this.
